sql/2020/14_PWA/top_manifest_orientations.sql

```diff
@@ -0,0 +1,38 @@
+#standardSQL
+# Top manifest orientations - based on 2019/14_04g.sql
+CREATE TEMPORARY FUNCTION getOrientation(manifest STRING)
+RETURNS STRING LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(manifest);
+  if (!('orientation' in $)) {
+    return '(not set)';
+  }
+  return $.orientation;
+} catch (e) {
+  return null;
+}
+''';
+
+SELECT
+  client,
+  getOrientation(body) AS orientation,
```

Comment:
Also consider an SQL-based approach (again, slight variation in results, but not significant)
```suggestion
  IFNULL(JSON_EXTRACT_SCALAR(body, '$.orientation'), '(not set)') AS orientation,
```

Reply:
Think I'll stick with UDF for consistency even though difference is small.